### PR TITLE
[Compute] `az sig image-definition create`: Add warning message for Hyper-V generation and Security Type

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

relate: https://github.com/Azure/azure-cli/issues/28557
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -4658,6 +4658,11 @@ def create_gallery_image(cmd, resource_group_name, gallery_name, gallery_image_n
                          minimum_cpu_core=None, maximum_cpu_core=None, minimum_memory=None, maximum_memory=None,
                          disallowed_disk_types=None, plan_name=None, plan_publisher=None, plan_product=None, tags=None,
                          hyper_v_generation='V1', features=None, architecture=None):
+    logger.warning(
+        "Please note that in the future, the default Hyper-V generation will be changed from V1 to V2 and "
+        "Security Type is set to “TrustedLaunchSuppoted” as the default behavior."
+    )
+
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:4665]:
```suggestion
    logger.warning(
        'Starting Build (May) 2024, "az sig image-definition create" command will use the new default values Hyper-V Generation: V2 and SecurityType: TrustedLaunchSuppoted.'
    )

```

**Comment by yanzhudd** [comment]
please convert this PR from draft to normal state

**Review by yanzhudd — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -4658,6 +4658,11 @@ def create_gallery_image(cmd, resource_group_name, gallery_name, gallery_image_n
                          minimum_cpu_core=None, maximum_cpu_core=None, minimum_memory=None, maximum_memory=None,
                          disallowed_disk_types=None, plan_name=None, plan_publisher=None, plan_product=None, tags=None,
                          hyper_v_generation='V1', features=None, architecture=None):
+    logger.warning(
+        "Starting Build (May) 2024, \"az sig image-definition create\" command will use the new default values "
+        "Hyper-V Generation: V2 and SecurityType: TrustedLaunchSuppoted."
```
Comment (johnl2323) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:4663]: @yanzhudd  @ZengTaoxu    "TrustedLaunchSuppoted" is spelled incorrectly.

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:4663]: Hi @johnl2323,
thanks for pointing out, we'll fix it.

**Pull request merged**